### PR TITLE
Fix if no segements are set

### PR DESCRIPTION
**Pull request by Quehnie**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| License | MIT

#### What's in this PR?

Fixed segments index missing in controller.

#### Why?

Update from sulu 2.0 to sulu 2.2.3 segments are not given.

#### To Do

- [ ] Create a documentation PR
- [ ] Add breaking changes to UPGRADE.md

### CODE REVIEW

**Comment by Quehnie** [comment]
was a mistake on our site overwriting the excerpt tab...

**Comment by alexander-schranz** [comment]
@nnatter how should we handle this. If somebody overide the excerpt and has the "segements" not in it `segements` are not set in the excerptdata. Currently it did just fail in the contentrouteprovider but not sure if there are other cases we need to test if the segments field type is removed from excerpt.

File: src/Sulu/Bundle/WebsiteBundle/Routing/ContentRouteProvider.php [hunk 1]
```diff
@@ -208,7 +208,7 @@ public function getRouteCollectionForRequest(Request $request)
                 return $collection;
             } else {
                 if ($document instanceof ExtensionBehavior) {
-                    $documentSegments = $document->getExtensionsData()['excerpt']['segments'];
```
Comment (niklasnatter) [inline src/Sulu/Bundle/WebsiteBundle/Routing/ContentRouteProvider.php:211]: please adjust this in the `Sulu\Bundle\RouteBundle\Routing\RouteProvider` too

**Comment by niklasnatter** [comment]
> Currently it did just fail in the contentrouteprovider but not sure if there are other cases we need to test if the segments field type is removed from excerpt.

How can you remove a field from the excerpt? wouldnt you just set the `visibleCondition` to `false`? 🤔 
That said, I think it should not hurt anybody to fallback to an empty array. So this is okay for me.

**Comment by alexander-schranz** [comment]
@nnatter excerpt override need to override the `excerpt.xml` in your  `pages` folder and if you don't add there the segments again in that excerpt xml it will totally get lost.

**Comment by alexander-schranz** [comment]
@nnatter we worked around this problem with `xinclude` here https://github.com/sulu/sulu-demo/pull/76/files#diff-e63b89f4647abb90d9c7fee3a8be28b6be23875590a74317a71a35a25ecb0417R16-R17
but @Quehnie did not use xinclude instead a whole copy of the excerpt.xml.

**Comment by niklasnatter** [comment]
> @nnatter excerpt override need to override the excerpt.xml in your pages folder and if you don't add there the segments again in that excerpt xml it will totally get lost.

You are right 🙂

File: src/Sulu/Bundle/WebsiteBundle/Routing/ContentRouteProvider.php (hunk 1 above)
Comment (alexander-schranz) [reply src/Sulu/Bundle/WebsiteBundle/Routing/ContentRouteProvider.php:211]: @nnatter did adjust it.

**Pull request merged**